src/components/core/PivotTable.tsx

```diff
@@ -1115,8 +1085,16 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
 
         invariant(execution !== undefined, "changing column width prior execution cannot work");
 
-        for (const column of columns) {
-            await this.resetResizedColumn(column);
+        if (this.isCtrlOrMetaKeyPressed() && columns.length === 1 && isMeasureColumn(columns[0])) {
+            this.resizedColumnsMap.removeAllMeasureColumns();
+            const measureColumnsToReset = this.columnApi.getAllColumns().filter(col => isMeasureColumn(col));
+            for (const column of measureColumnsToReset) {
+                await this.resetResizedColumn(column);
+            }
+        } else {
+            for (const column of columns) {
```

Comment:
the for is same in both cases, could you try to simplify this? maybe put the condition nad filtering into separate lambda or function?

Reply:
done